### ISSUE TITLE
How do I use `map_with`?

### ISSUE THREAD

**Issue by ManicMarrc** [zesterer/chumsky#545]
I have this piece of code:

```rust
fn stmt_parser<'src, I>() -> impl Parser<'src, I, Stmt, extra::Err<Rich<'src, Token>>>
where
  I: ValueInput<'src, Token = Token, Span = SimpleSpan>, {
  let expression = expr_parser()
    .then_ignore(just(Token::Semicolon))
    .map_with(|value, e| Stmt::Expression {
      value,
      span: e.span(),
    });

  expression
}
```

But I can't compile it, because "lifetime may not live long enough. argument requires that `'src` must outlive `'static`". How do I fix this?

**Comment by zesterer** [comment]
Argh. This is not supposed to happen. Looking into this now...

**Comment by zesterer** [comment]
Ok, I've just merged a fix in #546. Could you check to see whether this issue is fixed? If you're currently pointing at the latest alpha release you can point to `chumsky { git = "https://github.com/zesterer/chumsky.git" }` instead.

**Comment by ManicMarrc** [comment]
I can confirm that it is now fixed!

**Comment by zesterer** [comment]
Good good, I was worried for a moment that this was an unresolvable issue with the API!

**Comment by zesterer** [comment]
I've released `1.0.0-alpha.6` which includes this fix.
